assets/app/view/buy_trains.rb

```diff
@@ -9,6 +9,56 @@ module View
   class BuyTrains < Snabberb::Component
     include Actionable
     needs :show_other_players, default: nil, store: true
+    needs :selected_corporation, default: nil, store: true
+
+    def can_buy_others
+      !@game.actions.last.is_a?(Engine::Action::SellShares)
+    end
+
+    def render_president_contributions
+      player = @corporation.owner
+      funds_required = @depot.min_depot_price - (@corporation.cash + player.cash)
+      raisable = player.shares_by_corporation.sum do |corporation, shares|
```

Comment:
i think there's a new method called liquidity

Reply:
Yeah, unfortunately it counts shares that would cause the presidency to shift which aren't allowed in '89

Reply:
can that be added as an option to the call?